lib/samples/show_viewshed_from_geoelement_in_scene/show_viewshed_from_geoelement_in_scene.dart

```diff
@@ -0,0 +1,288 @@
+  Timer? _animationTimer;
+
+  // Waypoint for tank graphic.
+  ArcGISPoint? _waypoint;
+
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            ArcGISSceneView(
+              controllerProvider: () => _sceneViewController,
+              onSceneViewReady: onSceneViewReady,
+              onTap: onTap,
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+            // Banner at the top
+            SafeArea(
+              child: IgnorePointer(
+                child: Container(
+                  width: double.infinity,
+                  padding: const EdgeInsets.all(10),
+                  color: Colors.black.withValues(alpha: 0.5),
+                  child: const Text(
+                    'Tap on the map to move the tank and update the viewshed.',
+                    textAlign: TextAlign.center,
+                    style: TextStyle(color: Colors.white, fontSize: 16),
+                  ),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  // Called when the scene view is ready.
+  Future<void> onSceneViewReady() async {
+    // Create and configure the scene with elevation.
+    final scene = _createScene();
+    // Assign the scene to the scene view controller.
+    _sceneViewController.arcGISScene = scene;
+
+    // Load the tank graphic from the local data.
+    _tankGraphic = await _loadTankGraphic();
+
+    // Add the tank graphic to the scene.
+    await _addTankToScene(_tankGraphic!);
+
+    // Set up the orbit camera controller to follow the tank.
+    _setupCameraController(_tankGraphic!);
+
+    // Add the viewshed to the scene.
+    await _addViewShedToScene(_tankGraphic!);
+
+    setState(() => _ready = true);
+  }
+
+  // Creates a scene with an imagery basemap and adds elevation data.
+  ArcGISScene _createScene() {
+    final scene = ArcGISScene.withBasemapStyle(BasemapStyle.arcGISImagery);
+
+    // Add world elevation source to the scene's surface.
+    final elevationSource = ArcGISTiledElevationSource.withUri(
+      Uri.parse(
+        'https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer',
+      ),
+    );
+    scene.baseSurface.elevationSources.add(elevationSource);
+
+    // Create the building layer and add it to the scene.
+    final buildingsLayer = ArcGISSceneLayer.withUri(
+      Uri.parse(
+        'https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/Buildings_Brest/SceneServer/layers/0',
+      ),
+    );
+    scene.operationalLayers.add(buildingsLayer);
+
+    return scene;
+  }
+
+  // Handle the on tap function.
```

Comment:
This comment could be more specific. Maybe like "Convert the tapped location to a waypoint in the scene and begin the tank animation to the waypoint."